psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c

```diff
@@ -0,0 +1,273 @@
+#include <project.h>
+
+#include "dht11.h"
+#include <stdint.h>
+#include <stddef.h>
+#include <dht.h>
+
+void delay(void);
+void unpack_dht_data(void);
+
+uint8_t g_check;
+uint8_t g_decode_flag;
+uint8_t g_decoded_temperature_data;
+uint8_t g_decoded_humidity_data;
+
+int g_temperature_values[8];
+int g_humidity_values[8];
+int g_diff_count_values[44] = {0};
+int g_countvalues_datatimer[44] = {0};
+int g_executed;
+int g_count;
+int g_risingedge_counts;
+int g_has_been_executed;
+
+_CY_ISR_PROTO(Delay_Timer_ISR_Handler);
+_CY_ISR_PROTO(DHT_Pin_ISR_Handler);
+
+// What Return_Decoded_Temperature does:
+//     - This is used to pass the decoded temp data out 
+//       through a call in dht11.h
+//
+// Parameters: None
+// Returns: - Data in g_decoded_temperature_data
+
+uint8_t dht_get_temperature(){
+    
+    if(g_decoded_temperature_data == NULL) {
+        return -1;
+    }
+    else{
+        return g_decoded_temperature_data; 
+    }    
+}    
+
+// What Return_Decoded_Humidity does:
+//    -Returns the data in g_decoded_humidity_data
+// 
+// Parameters: None
+// Returns: uint8_t g_decoded_humidity_data
+
+uint8_t dht_get_humidity(){
+    
+    if(g_decoded_humidity_data == NULL) {
+        return -1;
+    }else{
+        return g_decoded_humidity_data;
+    }    
+}   
+
+
+// What delay_funct does:
+//    - Enable Delay_Timer_ISR to detect Delay_Timer Terminal Count interrupt
+//    - Start the Delay_Timer component.
+//    - This function does not use the CPU delay API, CyDelay(),
+//        but uses Delay_Timer to generate delay. 
+//
+// Parameters: None
+// Return: None
+
+void delay_funct(void){
+    g_count = 1;
+    g_executed = 0;
+    Delay_Timer_ISR_Enable();
+    // Write 0 to Control_Reg_Delay_Timer to starting Delay_Timer component 
+    Control_Reg_Delay_Timer_Write(0);
+}    
+
+
+// What decode dht data does:
+//    - The value stored in g_countvalues_datatimer[] array is used to decode
+//      the data of DHT output
+//    - The time difference between g_countvalues_datatimer[i]
+//      and g_countvalues_datatimer[i-1] is calculated
+//    - If the g_countvalues_datatimer[i] is approximately 70 counts, it is data
+ //     0, and if g_countvalues_datatimer[i] is approximately 120, it is data 1. 
+//
+// Parameters: None
+// Returns: None
+
+void decode_dht_data(void){
+    int i;
+    for(i=0; i<44; ++i) {
```

Comment:
Spaces around "=" and "<". Also, what is 44? This should at least have a comment. Better yet, make it a named constant and stick it at the top of the file.